### PR TITLE
Bugfix/lattice 2102 bidirection association type check

### PR DESCRIPTION
**Pull request by you-leee**

### CODE REVIEW

File: src/test/java/com/openlattice/rehearsal/SetupTestData.kt [hunk 1]
```diff
@@ -77,7 +77,7 @@ open class SetupTestData : MultipleAuthenticatedUsersBase() {
          */
         fun checkLinkingFinished( importedGeneralPersonFqns: Set<String>): Boolean {
             val finishedEntitySets = realtimeLinkingApi.linkingFinishedEntitySets
-            val finished = importedGeneralPersonFqns.all { finishedEntitySets.contains(entitySetsApi.getEntitySetId(it)) }
+            val finished = importedGeneralPersonFqns.all { finishedEntitySets.contains(edmApi.getEntitySetId(it)) }
```
Comment (katherinebernstein) [inline src/test/java/com/openlattice/rehearsal/SetupTestData.kt]: shouldn't this be entitySetsApi ?

File: src/test/java/com/openlattice/rehearsal/data/DataControllerTest.kt [hunk 2]
```diff
@@ -344,7 +344,7 @@ class DataControllerTest : MultipleAuthenticatedUsersBase() {
             Assert.fail("Should have thrown Exception but did not!")
         } catch (e: UndeclaredThrowableException) {
             Assert.assertTrue(e.undeclaredThrowable.message!!
-                    .contains("differs from allowed entity types in association type ${edge1.id}", true))
+                    .contains("differs from allowed entity types ([]) in association type of entity set ${esEdge1.id}", true))
```
Comment (katherinebernstein) [inline src/test/java/com/openlattice/rehearsal/data/DataControllerTest.kt:347]: Not sure what the right way to do this is, but I feel like at some point we should avoid matching on the exact error string to allow for flexibility in the messages. But it does seem valuable to be able to compare the specific ids / fields that are causing the errors.

File: src/test/java/com/openlattice/rehearsal/edm/EdmTest.kt [hunk 3]
```diff
@@ -60,10 +60,10 @@ class EdmTest : MultipleAuthenticatedUsersBase() {
         val es = createEntitySet(EdmTestConstants.personEt)
 
         entitySetsApi.addEntitySetsToLinkingEntitySet(linkingEs.id, setOf<UUID>(es.id))
-        Assert.assertEquals(es.id, entitySetsApi.getEntitySet(linkingEs.id).linkedEntitySets.single())
+        Assert.assertEquals(es.id, edmApi.getEntitySet(linkingEs.id).linkedEntitySets.single())
```
Comment (katherinebernstein) [inline src/test/java/com/openlattice/rehearsal/edm/EdmTest.kt]: These should also be entitySetsApi I think

Comment (you-leee) [reply src/test/java/com/openlattice/rehearsal/edm/EdmTest.kt]: Yeah, the wrong version got merged

File: src/test/java/com/openlattice/rehearsal/SetupTestData.kt (hunk 1 above)
Comment (you-leee) [reply src/test/java/com/openlattice/rehearsal/SetupTestData.kt]: Same here, the wrong version got merged. Pushed the fix

**Review by katherinebernstein — DISMISSED** [review]

**Review by katherinebernstein — APPROVED** [review]

**Pull request merged**